app/models/message.rb

```diff
@@ -201,7 +201,17 @@ def email_notifiable_message?
   end
 
   def valid_first_reply?
-    outgoing? && human_response? && not_created_by_automation? && !private?
+    return false if !outgoing? || private?
+
+    return false if conversation.first_reply_created_at.present?
+
+    return false unless human_response?
+    return false if conversation.messages.outgoing
+                                .where.not(sender_type: 'AgentBot')
+                                .where.not(private: true)
+                                .where("(additional_attributes->'campaign_id') is null").count > 1
+
+    true
```

Comment:
```rb
def valid_first_reply?
  return false unless outgoing? && human_response? && !private?
  return false if conversation.first_reply_created_at.present?
  return false if conversation.messages.outgoing
                              .where.not(sender_type: 'AgentBot')
                              .where.not(private: true)
                              .where("(additional_attributes->'campaign_id') is null").count > 1

  true
end
```

This implementation can reduce the cognitive complexity error from codeclimate

Reply:
Not a fan of unless and ! in the same line. However, to keep the linter happy, I'll update this.